### PR TITLE
[all] Upgrade Gophercloud to v2

### PR DESCRIPTION
**Pull request by pierreprinetti**
**What this PR does / why we need it**:

This PR upgrades Gophercloud to v2. While this new version enables `context.Context` for cancellation and tracing, this PR does not tackle that just yet, but rather just enables that follow-up work.

**Special notes for reviewers**:
This is currently a bump to a release candidate, just to kick the tires and get some feedback from the CI. I will update this pull request to import `v2.0.0` once it's out. Thank you!

```release-note
NONE
```

### CODE REVIEW

**Comment by pierreprinetti** [comment]
/hold cancel
Tests pass 🎉
This is ready for review.

**Comment by pierreprinetti** [comment]
/hold

Everything going according to plan, we are releasing v2.0.0 on Monday. I will update the tag in `go.mod` as soon as it's done, and then we can merge if the maintainers are OK with it.

**Comment by pierreprinetti** [comment]
/hold cancel
Updated to Gophercloud v2.0.0, which has been released this morning.

PTAL!
/cc @zetaab @kayrus

**Review by zetaab — APPROVED** [review]
/approve

quite heavy PR, could not find anything wrong. As tests are passing good to go

File: pkg/openstack/loadbalancer.go
Comment (kayrus) [inline pkg/openstack/loadbalancer.go:1013]: it makes sense to define this func in `pkg/util`

Comment (pierreprinetti) [reply pkg/openstack/loadbalancer.go:1013]: Good point. I have replaced that with a call to `ptr.Deref` in k8s.io/utils

Comment (kayrus) [reply pkg/openstack/loadbalancer.go]: Even better. Thanks!

**Pull request merged**
